src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -32,18 +39,23 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
 
 @Component
 public class DrsService {
     private final Logger logger = LoggerFactory.getLogger("bio.terra.service.filedata.DrsService");
 
     private static final String DRS_OBJECT_VERSION = "0";
+    // atomic counter that we incr on request arrival and decr on request response
+    private static AtomicInteger currentDRSRequests = new AtomicInteger(0);
```

Comment:
If `DrsService` is a singleton component (which I think it is, since singleton is the spring default scope) then there's no need to make this `static`, as only one instance of `DrsService` will be created for the application context.